File: mycollab-services/src/main/java/com/esofthead/mycollab/events/SessionEvent.java
package com.esofthead.mycollab.events;

import com.esofthead.mycollab.eventmanager.ApplicationEvent;

public class SessionEvent {
	public static class UserProfileChangeEvent extends ApplicationEvent {
		private static final long serialVersionUID = 1L;

		private String fieldChange;

		public UserProfileChangeEvent(Object source, String fieldChange,
				Object data) {
			super(source, data);

			this.fieldChange = fieldChange;
		}

		public String getFieldChange() {
			return fieldChange;
		}
	}
}

File: mycollab-services/src/main/scala/com/esofthead/mycollab/events/SessionEvent.scala
package com.esofthead.mycollab.events

import com.esofthead.mycollab.eventmanager.ApplicationEvent

/**
  * @author MyCollab Ltd.
  * @since 5.0.3
  */
class SessionEvent {
}

object SessionEvent {

  class UserProfileChangeEvent(source: AnyRef, fieldChange: String, data: AnyRef) extends ApplicationEvent(source, data) {
    def getFieldChange: String = fieldChange
  }

}
